python/az/aro/azext_aro/custom.py

```diff
@@ -195,6 +195,20 @@ def aro_list_credentials(client, resource_group_name, resource_name):
     return client.list_credentials(resource_group_name, resource_name)
 
 
+def aro_list_admin_credentials(cmd, client, resource_group_name, resource_name):
+    # check for the presence of the feature flag and warn
+    # the check shouldn't block the API call - ARM can cache a feature state for several minutes
+    feature_client = get_mgmt_service_client(cmd.cli_ctx, ResourceType.MGMT_RESOURCE_FEATURES)
+    feature = feature_client.features.get(resource_provider_namespace="Microsoft.RedHatOpenShift",
+                                          feature_name="AdminKubeconfig")
+    accepted_states = [feature_client.features.models.SubscriptionFeatureRegistrationState.REGISTERED,
+                       feature_client.features.models.SubscriptionFeatureRegistrationState.REGISTERING]
+    if feature.properties.state not in accepted_states:
+        logger.warning("This operation requires the Microsoft.RedHatOpenShift/AdminKubeconfig feature to be registered")
+        logger.warning("To register run: az feature register --namespace Microsoft.RedHatOpenShift -n AdminKubeconfig")
+    return client.list_admin_credentials(resource_group_name, resource_name)
```

Comment:
Will it return to the stdout? How do we want users to interact with it? Redirect stdout to a file?